docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "            self.movie_id = int(movie_id)\n",
+    "            return  # Automatically skip this line\n",
+    "        user_id, rating, date_stamp, *_ = line.split(',')\n",
+    "        user_id, rating = int(user_id), int(rating)\n",
+    "        yield MovieView(user_id, self.movie_id, rating, date_stamp)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `ParseMovieViews` callable will be used to generate the initial reference `PCollection` of `MovieView` instances for the `Netflix-prize` dataset."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now work out a solution in a very similar fashion to parse _movie_titles_ as extracted from the `movie_titles.txt` dataset: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@beam.typehints.with_output_types(MovieTitle)  # type-hint annotation for the output\n",
+    "class ParseMovieTitles(beam.DoFn):\n",
+    "    \"\"\"ParDo main execution core. Each line of the DATA_FILE will be processed, \n",
+    "    and an instance of MovieTitle will be created and returned, accordingly.\"\"\"\n",
+    "    \n",
+    "    def process(self, line: str):\n",
+    "        from netflix_movies import MovieTitle\n",
+    "        line = line.strip()  # get rid of any useless tabulation\n",
+    "        if not line:\n",
+    "            return  # Skip Line\n",
+    "        movie_id, year, *name = line.split(',')  # title may contain a comma as well\n",
+    "        movie_id, name = int(movie_id), \",\".join(name)\n",
+    "        year = -1 if year == \"NULL\" else int(year)\n",
+    "        yield MovieTitle(movie_id, year, name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### **`S2.1`** The core `beam.Pipeline`\n",
+    "\n",
+    "Now it is finally the time to create the whole `beam.Pipeline` infrastructure, that will allow pluggable callables to be used (as implemented in the following Exercises).\n",
+    "The Pipeline will be also configured to leverage as much as possible parallel execution in a way that we could circumvent the Python [**GIL**](https://wiki.python.org/moin/GlobalInterpreterLock)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Generating PCollections**\n",
+    "\n",
+    "The first step will be to create a function that returns the reference `PCollection` we want to deal with. Having two separate **data objects** (i.e. `MovieView` and `MovieTitle`), we might assume we will be needing two separate functions to gather instances of either type."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def netflix_movie_views_collection(p: beam.Pipeline, data_file: str = DATA_FILE) -> beam.PCollection[MovieView]:\n",
```

Comment:
Please combine both functions in one cell (it's better for the user to run less cells) and

could you please add
#@title ....
and hide the code by default (it might be too much information for the user)